### PR TITLE
Refactoring Path Management in storage_backend

### PR DESCRIPTION
**Pull request by fvaleye**
Proposal for better handling path in a platform-safe manner (#39, #40)

### CODE REVIEW

File: rust/src/storage/file.rs [hunk 1]
```diff
@@ -17,6 +18,19 @@ impl FileStorageBackend {
...
+        &self,
+        path: &str,
+        path_to_join: &str,
+    ) -> String {
+        let new_path = Path::new(path);
+        new_path
+            .join(path_to_join)
+            .into_os_string()
+            .into_string()
+            .unwrap()
```
Comment (rtyler) [inline rust/src/storage/file.rs:31]: It's hard for me to ever imagine what sequence of events could lead to this `into_string()` call failing, but I will defer to @houqp's judgement on whether `join_path` should propagate its Error

**Review by rtyler — COMMENTED** [review]
By and large looks good to me, thanks for the pull request!

File: rust/src/storage/file.rs (hunk 1 above)
Comment (houqp) [reply rust/src/storage/file.rs:31]: i think it should be fairly safe here since we only have `&str` as inputs.

**Review by houqp — APPROVED** [review]
LGTM!

**Pull request merged**